File: SPCode/Reporting/Support/SPReportSettings.vb
Imports MT.Common.Controls.OutlookStyleNavigateBar
Imports System.ComponentModel

Namespace Reporting
    Public Class SPReportSettings

        Private nvbFilterView As NavigateBarButton
        Private nvbGroupingView As NavigateBarButton
        Private _ViewerForm As SPReportViewer
        Private _reportFilters As SPReportFilterCollection = Nothing
        Private _reportGroupings As SPReportGroupingCollection = Nothing
        Private _reportParameters As SPReportParameterCollection = Nothing
        Private _AdvancedSearchHandler As SPAdvancedSearchDelegate
        Private _FilterSetupHandler As SPFilterSetupDelegate

#Region "Properties"
        Protected Friend ReadOnly Property Filters() As SPReportFilterCollection
            Get
                If _reportFilters Is Nothing Then _reportFilters = New SPReportFilterCollection
                Return _reportFilters
            End Get
        End Property
        Protected Friend ReadOnly Property Groups() As SPReportGroupingCollection
            Get
                If _reportGroupings Is Nothing Then _reportGroupings = New SPReportGroupingCollection
                Return _reportGroupings
            End Get
        End Property
        Protected Friend ReadOnly Property Parameters() As SPReportParameterCollection
            Get
                If _reportParameters Is Nothing Then _reportParameters = New SPReportParameterCollection
                Return _reportParameters
            End Get
        End Property
        Public Property AdvancedSearchHandler() As SPAdvancedSearchDelegate
            Get
                Return _AdvancedSearchHandler
            End Get
            Set(ByVal value As SPAdvancedSearchDelegate)
                _AdvancedSearchHandler = value
            End Set
        End Property
        Public Property FilterSetupHandler() As SPFilterSetupDelegate
            Get
                Return _FilterSetupHandler
            End Get
            Set(ByVal value As SPFilterSetupDelegate)
                _FilterSetupHandler = value
            End Set
        End Property


        ''' <summary>
        ''' Handle to the Report Viewer Form to enable Settings Form refresh a Loaded Report.
        ''' </summary>
        ''' <value></value>
        ''' <remarks></remarks>
        <Description("Handle to the Report Viewer Form to enable Settings Form refresh a Loaded Report.")> _
        Protected Friend WriteOnly Property ViewerForm() As SPReportViewer
            Set(ByVal value As SPReportViewer)
                _ViewerForm = value
            End Set
        End Property
#End Region

#Region "Methods"
        Public Sub AddFilter(ByVal DisplayMember As String, ByVal ValueMember As String, ByVal [Type] As SPReportFilterFieldTypes)
            Filters.Add(DisplayMember, ValueMember, [Type])
        End Sub
        Public Sub AddGrouping(ByVal DisplayMember As String, ByVal ValueMember As String)
            Groups.Add(DisplayMember, ValueMember)
        End Sub
        Public Sub AddParameter(ByVal Name As String, ByVal Value As Object)
            Parameters.Add(Name, Value)
        End Sub
#End Region


        Private Sub InitializeNavBar()
            AppNavigation.CollapsibleScreenWidth = 150
            AppNavigation.Dock = DockStyle.Fill
            nvbFilterView = New NavigateBarButton()
            With nvbFilterView
                .RelatedControl = FilterView
                .Caption = "Report Filters"
                .ToolTipText = "Report Filter Options"
                .Image = My.Resources.Clock
                .Enabled = True
                .Key = "FILTERVIEW"
            End With

            nvbGroupingView = New NavigateBarButton
            With nvbGroupingView
                .RelatedControl = Nothing
                .Caption = "Report Grouping"
                .ToolTipText = "Report Grouping Options"
                .Image = My.Resources.Clock
                .Enabled = True
                .Key = "GROUPINGVIEW"
            End With

            AppNavigation.NavigateBarButtons.Clear()
            AppNavigation.NavigateBarButtons.Add(nvbFilterView)
            AppNavigation.NavigateBarButtons.Add(nvbGroupingView)

            AppNavigation.NavigateBarDisplayedButtonCount = 2
            AppNavigation.Theme = NavigateBarTheme.VS2005Color

        End Sub
        Private Sub InitializeFilters()
            FilterView.Filters = Filters

        End Sub

        Private Sub InitializeGrouping()

        End Sub

        Private Sub SPReportSettings_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load
            InitializeFilters()
            InitializeGrouping()
            InitializeNavBar()
            If Me._ViewerForm IsNot Nothing Then
                Me._ViewerForm.MdiParent = Me.MdiParent
                Me._ViewerForm.Show()
            End If

            Me.DockState = WeifenLuo.WinFormsUI.DockState.DockLeft

        End Sub
    End Class
End Namespace

File: SoftLogic.Core/Win/Reporting/Support/SPReportSettings.cs
using System.Text.RegularExpressions;
using System.Diagnostics;
using System;
using System.Drawing.Drawing2D;
using System.Windows.Forms;
using System.Collections;
using System.Drawing;
using Microsoft.VisualBasic;
using System.Data.SqlClient;
using System.Data;
using System.Collections.Generic;
using WeifenLuo.WinFormsUI;
using Microsoft.Win32;
using WeifenLuo;
using System.ComponentModel;
using SoftLogik.Win.UI.Controls.OutlookStyleNavigateBar;


namespace SoftLogik.Win
{
	namespace Reporting
	{
		public partial class SPReportSettings
		{
			public SPReportSettings()
			{
				InitializeComponent();
			}
			
			private NavigateBarButton nvbFilterView;
			private NavigateBarButton nvbGroupingView;
			private SPReportViewer _ViewerForm;
			private SPReportFilterCollection _reportFilters = null;
			private SPReportGroupingCollection _reportGroupings = null;
			private SPReportParameterCollection _reportParameters = null;
			private SPAdvancedSearchDelegate _AdvancedSearchHandler;
			private SPFilterSetupDelegate _FilterSetupHandler;
			
			#region Properties
			protected internal SPReportFilterCollection Filters
			{
				get
				{
					if (_reportFilters == null)
					{
						_reportFilters = new SPReportFilterCollection();
					}
					return _reportFilters;
				}
			}
			protected internal SPReportGroupingCollection Groups
			{
				get
				{
					if (_reportGroupings == null)
					{
						_reportGroupings = new SPReportGroupingCollection();
					}
					return _reportGroupings;
				}
			}
			protected internal SPReportParameterCollection Parameters
			{
				get
				{
					if (_reportParameters == null)
					{
						_reportParameters = new SPReportParameterCollection();
					}
					return _reportParameters;
				}
			}
			public SPAdvancedSearchDelegate AdvancedSearchHandler
			{
				get
				{
					return _AdvancedSearchHandler;
				}
				set
				{
					_AdvancedSearchHandler = value;
				}
			}
			public SPFilterSetupDelegate FilterSetupHandler
			{
				get
				{
					return _FilterSetupHandler;
				}
				set
				{
					_FilterSetupHandler = value;
				}
			}
			
			
			/// <summary>
			/// Handle to the Report Viewer Form to enable Settings Form refresh a Loaded Report.
			/// </summary>
			/// <value></value>
			/// <remarks></remarks>
			[Description("Handle to the Report Viewer Form to enable Settings Form refresh a Loaded Report.")]protected internal SPReportViewer ViewerForm
			{
				set
				{
					_ViewerForm = value;
				}
			}
			#endregion
			
			#region Methods
			public void AddFilter(string DisplayMember, string ValueMember, SPReportFilterFieldTypes @Type)
			{
				Filters.Add(DisplayMember, ValueMember, @Type);
			}
			public void AddGrouping(string DisplayMember, string ValueMember)
			{
				Groups.Add(DisplayMember, ValueMember);
			}
			public void AddParameter(string Name, object Value)
			{
				Parameters.Add(Name, Value);
			}
			#endregion
			
			
			private void InitializeNavBar()
			{
				AppNavigation.CollapsibleScreenWidth = 150;
				AppNavigation.Dock = DockStyle.Fill;
				nvbFilterView = new NavigateBarButton();
				nvbFilterView.RelatedControl = FilterView;
				nvbFilterView.Caption = "Report Filters";
				nvbFilterView.ToolTipText = "Report Filter Options";
                nvbFilterView.Image = global::SoftLogik.Properties.Resources.Clock;
				nvbFilterView.Enabled = true;
				nvbFilterView.Key = "FILTERVIEW";
				
				nvbGroupingView = new NavigateBarButton();
				nvbGroupingView.RelatedControl = null;
				nvbGroupingView.Caption = "Report Grouping";
				nvbGroupingView.ToolTipText = "Report Grouping Options";
				nvbGroupingView.Image = global::SoftLogik.Properties.Resources.Clock;
				nvbGroupingView.Enabled = true;
				nvbGroupingView.Key = "GROUPINGVIEW";
				
				AppNavigation.NavigateBarButtons.Clear();
				AppNavigation.NavigateBarButtons.Add(nvbFilterView);
				AppNavigation.NavigateBarButtons.Add(nvbGroupingView);
				
				AppNavigation.NavigateBarDisplayedButtonCount = 2;
				AppNavigation.Theme = NavigateBarTheme.VS2005Color;
				
			}
			private void InitializeFilters()
			{
				FilterView.Filters = Filters;
				
			}
			
			private void InitializeGrouping()
			{
				
			}
			
			private void SPReportSettings_Load(object sender, System.EventArgs e)
			{
				InitializeFilters();
				InitializeGrouping();
				InitializeNavBar();
				if (this._ViewerForm != null)
				{
					this._ViewerForm.MdiParent = this.MdiParent;
					this._ViewerForm.Show();
				}
				
				this.DockState = WeifenLuo.WinFormsUI.DockState.DockLeft;
				
			}
		}
	}
	
}
